docs/src/content/docs/getting-started/setup.md

```diff
@@ -0,0 +1,218 @@
+---
+title: Setup
+description: Explaining initial setup
+sidebar: 
+  order: 1
+---
+
+## Create new project from template
+
+You can use [degit](https://github.com/Rich-Harris/degit) package for quick start project with reatom.  
+
+##### Reatom + React + TypeScript + Prettier + Vite
+
+```
+npx degit github:artalar/reatom-react-ts my-project
+cd my-project
+
+npm install
+npm run dev
+```
+
+
+## Add to existing project
+
+```
+npm i @reatom/core
+```
+
+### With TypeScript
+
+You don't need to do anything, type inference works as you'd expect
+
+```ts
+// AtomMut<number>
+const numAtom = atom(3) 
+
+// AtomMut<string>
+const strAtom = atom('foo') 
+
+// Atom<string | number>
+const dynamicAtom = atom((ctx) => { 
+  const num = ctx.spy(numAtom)
+  const str = ctx.spy(strAtom)
+  return num > 0 ? num : str
+})
+```
+
+Generics supported as well
+
+```ts
+/* tsconfig: strictNullChecks: true */
+// AtomMut<string | null>
+const nullableAtom = atom<string | null>(null) 
+```
+
+You can play with this example on [typescript playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgQwMY2BAdgGhTCEOAXzgDMpC4ByAASgFNkCQB6VaB6gKF46wDO8LAFcQAQRZwAvPkIAKAMwBKANx9sQuEKiSqs5gupkIEamo2D4GMHqIGW8+egAeymQD5E3OHH5bRez8YFwA6ATAAT3lAu2UfP014HRlgsIjonTiEhOAyOBixOC8ABncEBN9GGBEoLDhA9V9SBgAbAQZvXyqGGrrtGCgE4m5iC25-YRFW1uQAI1aGO1TDEAAeHWAsAHM4AB8G6daPQpm1IA)
+
+### With ESlint
```

Comment:
MOVE IT TO SEPARATE GUIDLE PLEASE. Let's not add extra complexity for newbie right from the first steps.